apis/keda/v1alpha1/triggerauthentication_types.go

```diff
@@ -282,6 +285,29 @@ type AzureKeyVaultCloudInfo struct {
 	ActiveDirectoryEndpoint string `json:"activeDirectoryEndpoint"`
 }
 
+type GCPSecretManager struct {
+	Secrets []Secrets `json:"secrets"`
+	// +optional
+	Credentials *GCPCredentials `json:"gcpCredentials"`
+	// +optional
+	PodIdentity *AuthPodIdentity `json:"podIdentity"`
+}
+
+type GCPCredentials struct {
+	ClientSecret GCPSecretmanagerClientSecret `json:"clientSecret"`
+}
+
+type GCPSecretmanagerClientSecret struct {
+	ValueFrom ValueFromSecret `json:"valueFrom"`
+}
+
+type Secrets struct {
```

Comment:
nit: to be more consistent with the `Azure` implementation and avoiding name clashes with other providers
```suggestion
type GCPSecretManagerSecret struct {
```